src/mds/MDSDaemon.cc

```diff
@@ -376,10 +376,6 @@ void MDSDaemon::set_up_admin_socket()
 				     asok_hook,
 				     "Evict a client session by id");
   ceph_assert(r == 0);
-  r = admin_socket->register_command("session ls name=cap_dump,type=CephBool,req=false",
-				     asok_hook,
-				     "Enumerate connected CephFS clients");
-  ceph_assert(r == 0);
```

Comment:
```
client ls [--cap-dump] [<filters>...]                                          List client sessions based on a filter
session ls [--cap-dump] [<filters>...]                                         List client sessions based on a filter
```

these both do the same thing, isn't this redundant? Should' the `session ls [--cap-dump] [<filters>...] ` be removed instead of `session ls [--cap-dump]`?

Reply:
If the `[filter]` is optional, I don't think so.

Reply:
I'm now thinking if we really need two CLI cmds for the same thing? Can't we have something like `client_session ls` instead because even both are same implementation wise? https://github.com/ceph/ceph/blob/ea807834105141b6938ce6a9dba28667420efa2e/src/mds/MDSRank.cc#L2644-L2645

Reply:
> I'm now thinking if we really need two CLI cmds for the same thing? Can't we have something like `client_session ls` instead because even both are same implementation wise?
> 
> https://github.com/ceph/ceph/blob/ea807834105141b6938ce6a9dba28667420efa2e/src/mds/MDSRank.cc#L2644-L2645

It seems right to me. But I will wait for @batrick opinion as we need to think broadly on this, coz there wpould be many use cases of both commands and in tests too.

Reply:
The problem is we probably have many scripts in the wild depending on the existence of "client ls" or "session ls". We can't merge them into one command. We can however fix the "session ls" command to take the "[filter]" argument.

Reply:
> We can however fix the "session ls" command to take the "[filter]" argument.

but the existing `session ls` does accept `filter` arg right? Or is it something else you're pointing at?

Reply:
No. The whole point of the ticket is that the other `session ls` command description without the `[filter]` argument overrides the one that has it.

Reply:
ah, now i understand your point. makes sense, thanks patrick!